### ISSUE TITLE
testing on only local resolvable handles

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2SAFE/B2HANDLE-Migration#2]
Hi,

I have a local handle server which is only locally resolvable. This causes a lot of:
```
Warning: Handle 842/18c1630c-02d6-11e6-813e-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18c43d6c-02e9-11e6-824a-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18ca1310-02ce-11e6-86cc-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18cdf646-0293-11e6-bcef-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18d18e76-0295-11e6-bc21-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18d2c3c0-02ec-11e6-9c7c-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18d3d018-02a0-11e6-9ea3-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18d7a954-a8b4-11e5-849e-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18d8647e-02dd-11e6-a018-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18ddc172-02dc-11e6-9dcc-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18de0488-0297-11e6-baf5-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18e2bed2-02a2-11e6-b59b-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18e6b74e-02ca-11e6-a938-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18eae314-029d-11e6-bd19-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18eedfbc-02e5-11e6-add8-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/18fb5334-02c0-11e6-b331-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/1900e35e-02d3-11e6-a484-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/190aca3a-02d9-11e6-bab1-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
Warning: Handle 842/190fa89e-a8a6-11e5-a711-fa163e62896a contains a PPID entry pointing to a non-existing Handle!
```
Is it possible to give a `hostname:port` so we can use a local handle server or the handle server without going through `hdl.handle.net`

**Comment by TobiasWeigel** [comment]
Fixed in 1c16c4b3b7acdd7bbae47af6153923f178129ff1: The script now has a --handleserverurl param.

**Comment by cookie33** [comment]
Hi,

Thanks for the fix. It fails as follows. That is to bad.
```
Traceback (most recent call last):
  File "/home/handle1/git/B2HANDLE-Migration/b2handle-migration/migrationtool.py", line 298, in <module>
    migration_tool.execute()
  File "/home/handle1/git/B2HANDLE-Migration/b2handle-migration/migrationtool.py", line 48, in execute
    self.migrate_handles()
  File "/home/handle1/git/B2HANDLE-Migration/b2handle-migration/migrationtool.py", line 184, in migrate_handles
    original_record = self.retrieve_handle_record_remotely(predecessor)
  File "/home/handle1/git/B2HANDLE-Migration/b2handle-migration/migrationtool.py", line 90, in retrieve_handle_record_remotely
    return self.b2handleclient.retrieve_handle_record(handle_name)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 319, in retrieve_handle_record
    handlerecord_json = self.__get_handle_record_if_necessary(handle, handlerecord_json)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 1091, in __get_handle_record_if_necessary
    handlerecord_json = self.retrieve_handle_record_json(handle)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 298, in retrieve_handle_record_json
    response=response
b2handle.handleexceptions.GenericHandleError: Error during interaction with Handle Server (retrieving).
	Handle: 21.T12996/4D9847A8-AD8F-11E6-97E1-5254000DF0ED
	URL: http://epic3.storage.surfsara.nl:8001/api/handles/21.T12996/4D9847A8-AD8F-11E6-97E1-5254000DF0ED
	HTTP Status Code: 400
	Response: {"responseCode":301,"message":"That prefix doesn't live here","handle":"21.T12996/4D9847A8-AD8F-11E6-97E1-5254000DF0ED"}
```
I will rerun it without the local URL filled in..
